walkthroughs/ecs-service-connectivity/yelb/deployments/platformdeployment/AWS/ECS/yelb-cloudformation-ECS-AppMesh-deployment.yaml

```diff
@@ -0,0 +1,631 @@
+                      Retries: 10
+                  LogConfiguration:
+                      LogDriver: awslogs
+                      Options:
+                          awslogs-group: !Ref AWS::StackName
+                          awslogs-region: !Ref AWS::Region
+                          awslogs-stream-prefix: yelb-gateway
+                  Environment:
+                      - Name: ENVOY_LOG_LEVEL
+                        Value: debug
+                      - Name: ENABLE_ENVOY_STATS_TAGS
+                        Value: '1'
+                      - Name: ENABLE_ENVOY_DOG_STATSD
+                        Value: '1'
+                      - Name: STATSD_PORT
+                        Value: '8125'
+                      - Name: APPMESH_VIRTUAL_NODE_NAME
+                        Value:
+                            Fn::Join:
+                                - ''
+                                -
+                                    - mesh/
+                                    - !Ref Mesh
+                                    - /virtualGateway/
+                                    - 'yelb-gateway'
+                                    
+    YelbDb:
+        Type: 'AWS::RDS::DBCluster'
+        Properties:
+            MasterUsername: !Ref 'Username'
+            MasterUserPassword: !Ref 'Password'
+            DatabaseName: 'yelbdatabase'
+            DBClusterIdentifier: yelb-cluster
+            Engine: aurora-postgresql
+            EngineVersion: '10.7'
+            Port: !Ref 'DBPort'
+            DBClusterParameterGroupName: default.aurora-postgresql10
+            EnableCloudwatchLogsExports:
+                - postgresql
+            VpcSecurityGroupIds: [!Ref 'YelbDbSecurityGroup' ]
+                
+    YelbDbInstance1:
+        Type: 'AWS::RDS::DBInstance'
+        Properties:
+            DBInstanceIdentifier: yelb-cluster-instance1
+            Engine: aurora-postgresql
+            DBClusterIdentifier: !Ref YelbDb
+            PubliclyAccessible: 'true'
+            DBInstanceClass: db.r4.large
+            
+    YelbRedis:
+        Type: 'AWS::ElastiCache::CacheCluster'
+        Properties:
+            AutoMinorVersionUpgrade: 'true'
+            Engine: redis
+            CacheNodeType: cache.t2.micro
+            Port: !Ref 'RedisPort'
+            NumCacheNodes: '1'
+            VpcSecurityGroupIds: [!Ref 'YelbRedisServerSecurityGroup' ]
+
+    ServiceYelbUi:
+        Type: AWS::ECS::Service
+        Properties:
+            LaunchType: !Ref LaunchType
+            Cluster: !Ref Cluster
+            DesiredCount: !Ref CountOfUiTasks
+            ServiceRegistries:
+                - RegistryArn: !GetAtt YelbUiServiceDiscoveryEntry.Arn
+            TaskDefinition: !Ref 'TaskDefinitionYelbUi'
+            NetworkConfiguration:
+                AwsvpcConfiguration:
+                    AssignPublicIp: !Ref PublicIP
+                    Subnets: [ !Ref 'PublicSubnetOne' , !Ref 'PublicSubnetTwo'  ]
```

Comment:
Given there is a load balancer, it does not appear the services themselves need to be deployed to a public subnet. We always want to demonstrate best practices in our examples. Can we move these to a private subnet?

Reply:
I recall testing with private subnets and it should work just fine. Given that we are asking the end users to provide the subnet value, I will change it to map to subnet.

Reply:
I just recalled having this conversation. The point was to showcase how we control traffic flow with appmesh components even when the services are deployed on public subnets.